roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/StreetSection.java

```diff
@@ -130,44 +191,39 @@ public void moveFirstCarToNextSection() {
         }
     }
 
-    /**
-     * addCar adds a car to the queue of the streetSection
-     *
-     * @param car The car to add.
-     */
     @Override
-    public void addCar(ICar car) {
-        carQueue.add(car);
+    public boolean carCouldEnterNextSection() {
+        throw new IllegalStateException("street section is not empty, but last car could not be determined");
     }
 
-    /**
-     * removes the first car of the queue and returns the first Car
-     *
-     * @return
-     */
-    @Override
-    public ICar removeFirstCar() {
-        return carQueue.poll();
-    }
+    private double getFreeSpace() {
+        updateAllCarsPositions();
 
-    @Override
-    public ICar getFirstCar() {
-        if (carQueue == null) {
-            throw new IllegalStateException("carQueue in section cannot be null");
+        ICar lastCar = getLastCar();
+        if (lastCar != null) {
+            double lastCarPosition = getCarPositions().get(lastCar);
+            return lastCarPosition - lastCar.getLength();
```

Comment:
why minus? if first car = last car and on exit point (= 0.0), then it would be negative?

Reply:
as we have discussed earlier, we have different streetsection concepts in our heads, i have following vision: when car enters the section it will be placed on position 0, the end of street section will have respectively position == streetsection.length. additionally car position describes were the "nose" of the car is - thats why we have to do a minus -> so we will get the amount of space behind the car.

however you are right about the "0.0" case, i will add additional Math.max() validation.